src/vendor/errors.jsx

```diff
@@ -1,154 +1,255 @@
+      c = Exception.wrap(cause);
     }
 
     this.template = t; // string descripbing the problem
     this.props = p; // object with properties used by template
     this.cause = c; // chained reason
+    this.stackOffset = stackOffset;
+  }
+
+  get trace() {
+    return parseStack(this.stack).slice(this.stackOffset);
   }
 
   toString() {
-    if (missing(this.template)) {
-      return this.cause.toString();
+    const output = [];
+
+    if (exists(this.template)) {
+      output.push(expand(this.template, this.props));
     }
 
-    let output = expand(this.template, this.props);
+    if (this.trace) {
+      output.push(`stacktrace`);
+      output.push(
+        ...this.trace.map(s => {
+          const output = ['    '];
+
+          if (exists(s.function)) output.push('at {{function}}');
+
+          if (exists(s.fileName)) {
+            output.push(' ({{fileName}}');
+
+            if (exists(s.line)) output.push(' line {{line}} column {{column}}');
+            output.push(')');
+          }
+
+          return expand(output.join(''), s);
+        })
+      );
+    }
 
     if (this.cause) {
-      output = `${output}\ncaused by\n${this.cause.toString()}`;
+      output.push('caused by');
+      output.push(this.cause.toString());
     }
 
-    return output;
+    return output.join('\n');
+  }
+
+  get message() {
+    // RETURN THE SHORT MESSAGE
+    const { template, props, cause } = this;
+
+    if (cause) return cause.message;
+
+    if (template) return expand(template, props);
+
+    return 'unknown error';
+  }
+
+  toData() {
+    return {
+      template: this.template,
+      props: this.props,
+      cause: exists(this.cause) ? this.cause.toData() : null,
+      trace: this.trace,
+    };
   }
 }
 
 Exception.wrap = err => {
-  if (err instanceof Exception) {
-    return err;
-  }
+  if (missing(err)) return null;
 
-  return new Exception(err);
+  if (err instanceof Exception) return err;
+
+  const output = new Exception();
+
+  output.template = err.message;
+  output.props = null;
+  output.cause = null;
+  output.stack = err.stack;
+
+  return output;
 };
 
 class Log {}
 
 Log.error = (template, params, cause) => {
-  throw new Exception(template, params, cause);
+  throw new Exception(template, params, cause, 1);
 };
 
 Log.warning = (template, params, cause) => {
-  const e = new Exception(template, params, cause);
+  let e = null;
+
+  if (template instanceof Error) {
+    e = Exception.wrap(template);
+  } else {
+    e = new Exception(template, params, cause, 1);
+  }
 
   // eslint-disable-next-line no-console
   console.log(e.toString());
 };
 
-class ErrorMessage extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
-
-  componentDidCatch(err, info) {
-    const error = Exception.wrap(err, info);
-
-    this.setState({ error });
-
-    Log.warning(err);
-  }
-
-  render() {
-    const { error } = this.state;
+const withErrorBoundary = WrappedComponent => {
```

Comment:
Why is `withErrorBoundary` in this module instead of its own?

Reply:
I'm also noticing a lot of new classes in here that I don't know what value they're adding (e.g. Exception.wrap)and we're lossing some important feature like reporting to Sentry [1] or reporting depending if we're on production vs development [2]

I would need a strong case to take changes that would make us loose this.

[1] https://github.com/mozilla-frontend-infra/firefox-health-dashboard/blob/13c943fb3c788647eaefe138e3f71368669cc930/src/components/genericErrorBoundary.jsx#L13
[2] https://github.com/mozilla-frontend-infra/firefox-health-dashboard/blob/13c943fb3c788647eaefe138e3f71368669cc930/src/hocs/withErrorBoundary.jsx#L6

Reply:
Also this wrapping things are causing them to change the normal behaviour of the console logging.
For instance, this error should not be an info message (it should show red):
<img width="1160" alt="image" src="https://user-images.githubusercontent.com/44410/54047659-a7139b00-41a5-11e9-83c4-7f1557dfa259.png">

Can you please move all of this logic affecting console logging to be moved into another PR so we can just focus on error boundaries? Or better than that, file an issue so we can understand why these changes would be needed. I would not want you to spend time creating another PR if there's no clear value to be gained.